src/helpers/factories.py

```diff
@@ -26,12 +27,17 @@ def problem_factory(problem: EProblems):
     return problem_classes[problem]()
 
 
-def adaptation_factory(adaptation: EAdaptations, base_points: torch.Tensor, x_range: [float, float],
-                       max_number_of_points: int=params.NUM_MAX_POINTS):
+def adaptation_factory(
```

Comment:
Sorry for this formatting changes. Do you use any linter/formatter? Wdyt about using pre-commit? I can set it up.

Reply:
I've used one before, but for this repo I've rewritten everything and I had no time to do multiple things I wanted 

Feel free to edit whatever you want :D